pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -331,7 +327,9 @@ public boolean isValidOriginalPrincipal(String authenticatedPrincipal,
             }
         } else if (StringUtils.isNotBlank(originalPrincipal)
                 && !(allowNonProxyPrincipalsToBeEqual && originalPrincipal.equals(authenticatedPrincipal))) {
-            errorMsg = "cannot specify originalPrincipal when connecting without valid proxy role.";
+                log.warn("[{}] Non-proxy role [{}] passed originalPrincipal [{}]. This behavior will not "
```

Comment:
I agreed with @mattisonchao. Remove the print log to avoid too many warnings.

Reply:
I think it's justified to log a warning since it's a true warning about invalid usage. The Pulsar admin should react and fix the invalid usage to get rid of the warning.

Reply:
@lhotari It should also affect the TCP connection. No?

Reply:
This log will appear anytime proxy and original roles are validated and meet the appropriate conditions, which is once for connections over the binary protocol and once for each admin http request.

I think it's a justified warning log since it is warning an administrator that their current configuration will not work in an upcoming release. Operators will only have to update their proxy's authentication to use a `proxyRole` and the log line will be removed in their current version.

> @lhotari It should also affect the TCP connection. No?

I think @lhotari means the Pulsar human administrator, not the admin service.

Reply:
I think it's fine to add a warning log here. It will help the pulsar's administrator to consider correcting the configuration. And the warning logs will also not impact the alert too much.

Reply:
Ok, I see your guys' point. Thanks for the explanation.